core/src/common/dtls_abstraction_tinydtls.c

```diff
@@ -222,14 +223,23 @@ bool DTLS_Encrypt(NetworkAddress * destAddress, uint8_t * plainText, int plainTe
         {
             session->UserContext = context;
             session->Callbacks.write = SSLSendCallBack;
+            dtls_peer_t * peer = dtls_get_peer(session->Context, &session->Session);
+            if (!peer)
+            {
+                dtls_connect(session->Context, &session->Session);
+            }
         }
     }
     else
     {
         session = AllocateSession(destAddress, true, context);
         if (session)
         {
-            dtls_connect(session->Context, &session->Session);
+            dtls_peer_t * peer = dtls_get_peer(session->Context, &session->Session);
```

Comment:
Hi Delme, it looks like you may have combined multiple pieces of work on this PR.